core/src/test/java/org/jclouds/proxy/ProxyForURITest.java

```diff
@@ -105,47 +112,62 @@ public void testUseProxyForSocketsSettingShouldntAffectHTTP() throws Exception {
       URI uri = new URI("https://example.com/file");
       assertEquals(new ProxyForURI(config).apply(uri), Proxy.NO_PROXY);
    }
 
    @Test
    public void testFTPDirect() throws URISyntaxException {
-      ProxyConfig config = new MyProxyConfig(false, Proxy.Type.DIRECT, noHostAndPort, noCreds);
+      ProxyConfig config = new MyProxyConfig(false, false, Proxy.Type.DIRECT, noHostAndPort, noCreds);
       URI uri = new URI("ftp://ftp.example.com/file");
       assertEquals(new ProxyForURI(config).apply(uri), Proxy.NO_PROXY);
    }
 
    @Test
    public void testSocketDirect() throws URISyntaxException {
-      ProxyConfig config = new MyProxyConfig(false, Proxy.Type.DIRECT, noHostAndPort, noCreds);
+      ProxyConfig config = new MyProxyConfig(false, false, Proxy.Type.DIRECT, noHostAndPort, noCreds);
       URI uri = new URI("socket://ssh.example.com:22");
       assertEquals(new ProxyForURI(config).apply(uri), Proxy.NO_PROXY);
    }
 
    @Test
    public void testHTTPThroughHTTPProxy() throws URISyntaxException {
-      ProxyConfig config = new MyProxyConfig(false, Proxy.Type.HTTP, hostAndPort, creds);
+      ProxyConfig config = new MyProxyConfig(false, false, Proxy.Type.HTTP, hostAndPort, creds);
       URI uri = new URI("http://example.com/file");
       assertEquals(new ProxyForURI(config).apply(uri), new Proxy(Proxy.Type.HTTP, new InetSocketAddress(
             "proxy.example.com", 8080)));
    }
 
    @Test
-   public void testHTTPThroughSystemProxy() throws URISyntaxException {
-      ProxyConfig config = new MyProxyConfig(true, Proxy.Type.DIRECT, noHostAndPort, noCreds);
+   public void testThroughJvmProxy() throws URISyntaxException {
+      ProxyConfig config = new MyProxyConfig(false, true, Proxy.Type.HTTP, noHostAndPort, noCreds);
       URI uri = new URI("http://example.com/file");
-      // could return a proxy, could return NO_PROXY, depends on the tester's
-      // environment
+      // could return a proxy, could return NO_PROXY, depends on the tester's environment
       assertNotNull(new ProxyForURI(config).apply(uri));
    }
+   
+   @Test
+   public void testThroughSystemProxy() throws URISyntaxException {
+      ProxyConfig config = new MyProxyConfig(true, false, Proxy.Type.HTTP, noHostAndPort, noCreds);
+      URI uri = new URI("http://example.com/file");
+      // could return a proxy, could return NO_PROXY, depends on the tester's environment
+      assertNotNull(new ProxyForURI(config).apply(uri));
+   }
+
+   @Test
+   public void testJvmProxyAlwaysPreferredOverSystem() throws URISyntaxException {
+      ProxyConfig test = new MyProxyConfig(true, true, Proxy.Type.HTTP, noHostAndPort, noCreds);
+      ProxyConfig jvm = new MyProxyConfig(false, true, Proxy.Type.HTTP, noHostAndPort, noCreds);
+      URI uri = new URI("http://example.com/file");
+      assertEquals(new ProxyForURI(test).apply(uri), new ProxyForURI(jvm).apply(uri));
+   }
```

Comment:
Add a test that uses jvmProxy | systemProxy and also sets the jclouds proxy host? Just to verify that the jclouds proxy takes precedence.